api/glossary_usefulconcepts.md

```diff
@@ -0,0 +1,156 @@
+Glossary of useful concepts
+============================
+###### Narrowing the search space when you don’t even know what you don’t know
+
+## Backend technology stack
+This is divided into two sections. The technology stack is described first, as an introduction to the backend technologies used in Baobab. All words in **bold** are also defined within this document.
+
+### Python
+[Python](https://www.python.org/doc/essays/blurb/) is an interpreted object-orientated, high level programming language with dynamic semantics.
+Let’s break that up:
+
+**1. Interpreted versus compiled languages:**
+
+Languages can have implementations of both
+
+- Compiled:
+written code is converted directly into machine code that the processor executes
+More control over hardware (memory, CPU usage). Examples are C, C++, Go
+- Interpreted: 
+these languages will run through the programme line by line and execute each.
+Examples include JavaScript, Python
+
+**2. Object-orientated programming**
+- This centres around the creation of objects that are characterised by attributes and behaviour. 
+- **Classes** are used in OOP in the creation of objects. High-level versus low-level languages
+- This refers to the level of abstraction. Each level is a step away from the base machine 
+    - Abstraction: the degree to which the characteristics of an object are hidden in order to reduce complexity and increase efficiency. The remaining object is a representation of the original with unwanted detail omitted
+    - Higher level languages have greater levels of abstraction 
+
+**3. Dynamic semantics**
+- This defines how and when the various constructs of a programming language should produce a programme behaviour
+- Constructs refer to instances of values contained into so-called constructs which exist at runtime. This helps the flow of execution of the programme.
```

Comment:
No idea what this means :) I suggest removing it, maybe add "Dynamic Typing" instead which is more practically relevant.